### PR TITLE
Add collection:updateSpecifications

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

This PR implements the `collection:updateSpecifications` method with its unit tests.

### How should this be manually tested?

Clone this branch and run unit tests
`./gradlew test`

When it succeed, compile it

`./gradlew jar`

Initiate another java project by adding the compiled SDK as a dependency.

Then, run Kuzzle, create an index `nyc-open-data`, a `yellow-taxi` collection
Finally, run this code

```java
import io.kuzzle.sdk.*;
import io.kuzzle.sdk.Options.KuzzleOptions;
import io.kuzzle.sdk.Options.Protocol.WebSocketOptions;
import io.kuzzle.sdk.Protocol.WebSocket;

import java.util.concurrent.ConcurrentHashMap;

public class updateSpecificationsCollection {
    private static Kuzzle kuzzle;

    public static void main(String[] args) {
        WebSocketOptions opts = new WebSocketOptions();
        opts.setAutoReconnect(true).setConnectionTimeout(42000);

        try {
            WebSocket ws = new WebSocket("localhost", opts);

            kuzzle = new Kuzzle(ws, (KuzzleOptions) null);

            kuzzle.connect();
           
            ConcurrentHashMap<String, Object> specifications = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> fields = new ConcurrentHashMap<>();
            ConcurrentHashMap<String, Object> license = new ConcurrentHashMap<>();

            specifications.put("strict", false);
            license.put("mandatory", true);
            license.put("type", "string");
            fields.put("license", license);
            specifications.put("fields", fields);

            ConcurrentHashMap<String, Object> result = 
            kuzzle.getCollectionController().updateSpecifications("nyc-open-data", "yellow-taxi",  
            specifications).get();
           
        }  catch (Exception e) {
            e.printStackTrace();
        }

        kuzzle.disconnect();
    }
};
```

### CODE REVIEW

File: doc/3/controllers/collection/update-specifications/index.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+| `collection`     | <pre>String</pre>                            | Collection name          |
+| `specifications` | <pre>ConcurrentHashMap<String, Object></pre> | Specifications to update |
+
+### specifications
+
+A `ConcurrentHashMap<String, Object>` representing the specifications.
+
+It must follow the [Specification Structure](/core/2/guides/essentials/data-validation):
+
+```java
```
Comment (scottinet) [inline doc/3/controllers/collection/update-specifications/index.md:37]: The content of this example doesn't appear to be written in java.

Given that there is already a code snippet in java, and JSON examples in the link you put a few lines back, then I think that this example should be removed.

Comment (Aschen) [reply doc/3/controllers/collection/update-specifications/index.md:37]: I think it's the output of a `println` on the `HashMap` but still I agree with @scottinet, you should keep the JSON

File: doc/3/controllers/collection/update-specifications/snippets/update-specifications.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        .get();
+
+/*
+   {
+    strict=false,
+    fields={
+      license={
+        mandatory=true,
+        type="string"
+          }
```
Comment (Aschen) [inline doc/3/controllers/collection/update-specifications/snippets/update-specifications.java:23]:
```suggestion
      }
```

File: doc/3/controllers/collection/update-specifications/snippets/update-specifications.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        }
```
Comment (Aschen) [inline doc/3/controllers/collection/update-specifications/snippets/update-specifications.java:24]:
```suggestion
      }
```

**Pull request merged**